lib/Parse/ParseDecl.cpp

```diff
@@ -2588,6 +2589,8 @@ bool Parser::isStartOfDecl() {
   // If this is obviously not the start of a decl, then we're done.
   if (!isKeywordPossibleDeclStart(Tok)) return false;
 
+  if (peekToken().is(tok::colon)) return false;
```

Comment:
As I suggested in my other comment this should probably be - `if (Tok.canBeArgumentLabel() && peekToken().is(tok::colon)) return false;`.

Reply:
Safer change might be to add that logic to `parseList` when it's detected that token is at the newline and separator omission is allowed.

Reply:
In https://github.com/apple/swift/pull/22714/commits/faecb8d5f15a3421f661bd606158d0394dbb19be , the logic you described here is used and the change to `isStartOfStmt` and `isStartOfDecl` is reverted.